### ISSUE TITLE
Getting the accuracy scores

### ISSUE THREAD

**Issue by PrekshyaBasnet** [yashsmehta/personality-prediction#6]
@yashsmehta @saminfatehir 

Hi,

I wanted to know how the accuracy score was calculated for each traits. 

In the output explog, I get accuracy score for 10 fold for each traits. I am trying to reproduce the result from the paper, so I wanted to know are these scores for each traits calculate, are they averaged or chosen from the maximum score from the 10 accuracy scores?

Thanks 😊

**Comment by yashsmehta** [comment]
Hi @PrekshyaBasnet, 
Since the accuracy is reported on 10-fold cross-validation, the performance of a run is calculated as the _average_ of the scores from each of the folds. The final performance of a model is the average of 10 runs with different model initializations (seeds).

**Comment by PrekshyaBasnet** [comment]
@yashsmehta 
I am little bit confused with the model initialisations part. 

So for example, for Extraversion with default configuration, I got 10 accuracies from explog which I will have to average out, then by hyper-parameter tuning the parameters for epochs, batches, etc in the initialisation part, I will have to run the model again and average the scores. So the final performance would be average of the performance of each model run on 10 folds?

So if I have to save the optimal model I should use the average score of each model(10 fold) and choose the one with maximum accuracy right?

Thanks,
Prekshya

**Comment by yashsmehta** [comment]
@PrekshyaBasnet the hyperparameter tuning part would be separate. After you have the optimal hyperparams (e.g. LR), to get the average accuracy with that configuration, you would run your model (e.g. MLP) x times (in the paper, x=10), with different seeds but the same hyperparam config and get the scores for 10 folds and average over these x times to get the 'performance' of a model with particular hyperparam setting. For getting the optimal hyperparam config, you would similarly run models with different configurations and choose the hyperparams with the highest 'performance' as the optimal ones.

**Comment by PrekshyaBasnet** [comment]
@yashsmehta Thank you for you reply!

When you say seed what in particular are you mentioning? Can you direct me in the code implementation?

Is it the params that are set using the code below in the main function of MLP_LM like this: `  inp_dir, dataset, lr, batch_size, epochs, log_expdata, embed, layer, mode, embed_mode, jobid = utils.parse_args()`?

Sorry for the confusion I am trying to understand this. 

Thanks,
Prekshya

**Comment by yashsmehta** [comment]
In MLP_LM.py, you will find (line 119):

```
seed = jobid    
np.random.seed(seed)    
tf.random.set_seed(seed)
```
